Semantics.md

```diff
@@ -157,6 +162,146 @@ Store operators do not produce a value.
+| `i32.atomic_xor8_s` | 1 byte, sign-extended i8 to i32 | sign-agnostic exclusive or | 1 byte, wrapped from i32 to i8 |
+| `i32.atomic_xor8_u` | 1 byte, zero-extended i8 to i32 | sign-agnostic exclusive or | 1 byte, wrapped from i32 to i8 |
+| `i32.atomic_xor16_s` | 2 bytes, sign-extended i16 to i32 | sign-agnostic exclusive or | 2 bytes, wrapped from i32 to i16 |
+| `i32.atomic_xor16_u` | 2 bytes, zero-extended i16 to i32 | sign-agnostic exclusive or | 2 bytes, wrapped from i32 to i16 |
+| `i32.atomic_xor` | 4 bytes as i32 | sign-agnostic exclusive or | 4 bytes |
+| `i64.atomic_xor8_s` | 1 byte, sign-extended i8 to i64 | sign-agnostic exclusive or | 1 byte, wrapped from i64 to i8 |
+| `i64.atomic_xor8_u` | 1 byte, zero-extended i8 to i64 | sign-agnostic exclusive or | 1 byte, wrapped from i64 to i8 |
+| `i64.atomic_xor16_s` | 2 bytes, sign-extended i16 to i64 | sign-agnostic exclusive or | 2 bytes, wrapped from i64 to i16 |
+| `i64.atomic_xor16_u` | 2 bytes, zero-extended i16 to i64 | sign-agnostic exclusive or | 2 bytes, wrapped from i64 to i16 |
+| `i64.atomic_xor32_s` | 4 bytes, sign-extended i32 to i64 | sign-agnostic exclusive or | 4 bytes, wrapped from i64 to i32 |
+| `i64.atomic_xor32_u` | 4 bytes, zero-extended i32 to i64 | sign-agnostic exclusive or | 4 bytes, wrapped from i64 to i32 |
+| `i64.atomic_xor` | 8 bytes | sign-agnostic exclusive or | 8 bytes |
+| `i32.atomic_xchg8_s` | 1 byte, sign-extended i8 to i32 | nop | 1 byte, wrapped from i32 to i8 |
+| `i32.atomic_xchg8_u` | 1 byte, zero-extended i8 to i32 | nop | 1 byte, wrapped from i32 to i8 |
+| `i32.atomic_xchg16_s` | 2 bytes, sign-extended i16 to i32 | nop | 2 bytes, wrapped from i32 to i16 |
+| `i32.atomic_xchg16_u` | 2 bytes, zero-extended i16 to i32 | nop | 2 bytes, wrapped from i32 to i16 |
+| `i32.atomic_xchg` | 4 bytes as i32 | nop | 4 bytes |
+| `i64.atomic_xchg8_s` | 1 byte, sign-extended i8 to i64 | nop | 1 byte, wrapped from i64 to i8 |
+| `i64.atomic_xchg8_u` | 1 byte, zero-extended i8 to i64 | nop | 1 byte, wrapped from i64 to i8 |
+| `i64.atomic_xchg16_s` | 2 bytes, sign-extended i16 to i64 | nop | 2 bytes, wrapped from i64 to i16 |
+| `i64.atomic_xchg16_u` | 2 bytes, zero-extended i16 to i64 | nop | 2 bytes, wrapped from i64 to i16 |
+| `i64.atomic_xchg32_s` | 4 bytes, sign-extended i32 to i64 | nop | 4 bytes, wrapped from i64 to i32 |
+| `i64.atomic_xchg32_u` | 4 bytes, zero-extended i32 to i64 | nop | 4 bytes, wrapped from i64 to i32 |
+| `i64.atomic_xchg` | 8 bytes | nop | 8 bytes |
+
+The atomic compare exchange operators take three operands: a linear memory
+index, an expected value, and a replacement value. The operators conditionally
+store the replacement value, but only if the loaded value matches the expected
+value.
+
+| Name | Read | Modify | Write |
+| ---- | ---- | ---- | ---- |
+| `i32.atomic_cmpxchg8_s` | 1 byte, sign-extended i8 to i32 | nop | (conditional) 1 byte, wrapped from i32 to i8 |
```

Comment:
One detail that's not clear from this presentation is whether the comparison is the sign-extended value with the full 32 bits of the expected value, or the expected value wrapped to 8 bits with the 8-bit loaded value.